src/main/java/pt/tqs/g205/services/EncomendaRestauranteService.java

```diff
@@ -0,0 +1,66 @@
+package pt.tqs.g205.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import pt.tqs.g205.domain.Encomenda;
+import pt.tqs.g205.domain.EncomendaRestaurante;
+import pt.tqs.g205.domain.EstadoEncomenda;
+import pt.tqs.g205.domain.Prato;
+import pt.tqs.g205.domain.PratosPorEncomenda;
+import pt.tqs.g205.repositories.EncomendaRepository;
+import pt.tqs.g205.repositories.EncomendaRestauranteRepository;
+import pt.tqs.g205.repositories.EstadoEncomendaRepository;
+import pt.tqs.g205.repositories.PratoRepository;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Service
+public class EncomendaRestauranteService {
+  @Autowired
+  private EstadoEncomendaRepository estadoEncomendaRepo;
+  
+  @Autowired
+  private PratoRepository pratoRepo;
+  
+  @Autowired
+  private EncomendaRepository encomendaRepo;
+  
+  @Autowired
+  private EncomendaRestauranteRepository encomendaRestauranteRepo;
+  
+  /**
+   * Separar a encomenda em parcelas, com os produtos para cada restaurante respetivo.
+   * @param enc encomenda.
+   * @return lista de parcelas.
+   */
+  public List<EncomendaRestaurante> criarParcelas(Encomenda enc) {
+    List<EncomendaRestaurante> parcelas = new ArrayList<>();
+    
+    EstadoEncomenda estado = estadoEncomendaRepo.findById(1).get();
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Call "Optional#isPresent()" before accessing the value. ([squid:S3655](https://sonarcloud.io/organizations/bdomingues-github/rules#rule_key=squid%3AS3655))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=pt.tqs.g205%3Arestapi&pullRequest=20&issues=AWPIIg0LkDisRwWBeUf9&open=AWPIIg0LkDisRwWBeUf9)

[ISSUE_KEY]: AWPIIg0LkDisRwWBeUf9